airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -66,17 +66,27 @@ def replace_enum_allOf_and_anyOf(schema: Dict[str, Any]) -> Dict[str, Any]:
         for format in objects_to_check["oneOf"]:
             for key in format["properties"]:
                 object_property = format["properties"][key]
-                if "allOf" in object_property and "enum" in object_property["allOf"][0]:
-                    object_property["enum"] = object_property["allOf"][0]["enum"]
-                    object_property.pop("allOf")
+                AbstractFileBasedSpec.move_enum_to_root(object_property)
 
         properties_to_change = ["validation_policy"]
         for property_to_change in properties_to_change:
             property_object = schema["properties"]["streams"]["items"]["properties"][property_to_change]
             if "anyOf" in property_object:
                 schema["properties"]["streams"]["items"]["properties"][property_to_change]["type"] = "object"
                 schema["properties"]["streams"]["items"]["properties"][property_to_change]["oneOf"] = property_object.pop("anyOf")
-            if "allOf" in property_object and "enum" in property_object["allOf"][0]:
-                property_object["enum"] = property_object["allOf"][0]["enum"]
-                property_object.pop("allOf")
+            AbstractFileBasedSpec.move_enum_to_root(property_object)
+
+        csv_format_schemas = list(filter(lambda format: format["properties"]["filetype"]["default"] == "csv", schema["properties"]["streams"]["items"]["properties"]["format"]["oneOf"]))
```

Comment:
This is ugly as hell but without this, `header_definition` is defined as:

```
"header_definition": {
    "title": "Header Definition",
    "description": "How headers will be defined. `User Provided` assumes the CSV does not have a header row and users the headers provided and `Autogenerated` assumes the CSV does not have a header row and the CDK will generate headers using for `f{i}` where `i` is the index starting from 0. Else, the default behavior is to use the header from the CSV file. If a user wants to autogenerate or provide column names for a CSV having headers, he can skip rows",
    "default": {
        "definition_type": "From CSV", "column_names": None
    },
    "allOf": [
        {
            "title": "CsvHeaderDefinition",
            "type": "object",
            "properties": {
                "definition_type": {
                    "title": "CSV Header Definition Type",
                    "description": "How to infer column names.",
                    "allOf": [
                        {
                            "title": "CsvHeaderDefinitionType",
                            "description": "An enumeration.",
                            "enum": ["User Provided", "Autogenerated", "From CSV"]
                        }
                    ]
                },
                "column_names": {
                    "title": "Column Names",
                    "description": "Given type is user-provided, column names will be provided using this field",
                    "type": "array",
                    "items": {"type": "string"}
                }
            },
            "required": ["definition_type"]
        }
    ]
}
```

... and if I remember correctly, `allOf` are not supported

Reply:
Using oneOf has reduce a little bit the complexity here but the problem still stands. We will assume this is ok for now. Newer versions of pydantics allow us to manage schema more locally so we should eventually look forward to this